lib/core/versions/latest/UpdateOperation.ts

```diff
@@ -112,9 +113,12 @@ export default class UpdateOperation implements OperationModel {
       if (operationObject.type !== OperationType.Update) {
         throw new SidetreeError(ErrorCode.UpdateOperationTypeIncorrect);
       }
-
-      encodedDelta = operationObject.delta;
-      delta = await Operation.parseDelta(encodedDelta);
+      Operation.validateDelta(operationObject.delta);
+      delta = {
+        patches: operationObject.delta.patches,
+        updateCommitment: operationObject.delta.update_commitment
```

Comment:
this kind of casing makes me so sad :(

Reply:
@OR13, are you talking about the mix of snake casing and camel casing? If so, remember I fought hard on having just camel casing and even made the case that DID core spec is using camel casing in WG meeting back in like April or March and lost?

Reply:
yep, we all lost... the sidetree spec still does not have a "styleguide" which defines this.

I have already hit a bunch of casing related bugs because of this.

But I assume its too late to fix now.

Reply:
I cannot begin to express the level of frustration I faced when OTHERS supported the current casing, voted to use it, then later wanted to change it to something else, which I opposed simply for stability sake. But if people want to play faux-amnesia revisionist history games and willfully forget that, screw it, just change it now and break our client teams, but I'd like to set a meeting where you own the change, given you all agreed that the the current casing and later wavered after it was accepted and implemented. I'm getting our whole team on the call to witness the hat-in-hand mea culpa, because I'm tired of being the messenger for a group of people who can't vote on something and stick to it.

Reply:
https://github.com/decentralized-identity/sidetree/issues/805

I provided links to API styleguides (Google and Microsoft), you provided examples of APIs that don't use the styleguides I linked (GitHub and Slack).

Regardless of which "path we take" the WG needs a styleguide, or stuff like this is going to keep coming up.

At this point, if you are willing to use Google and Microsoft JSON API styleguides, I am ok admitting that I was wrong to ever compromise with you, and that I should never have accepted the casing convention you proposed.

I admit that changing this now will create a big headache, but IMO its worth having a style that matches google and miscrosoft.

I admit that I did accept snake case, and that I am now rejecting it

I also want to point out that sidetree has not hit a 1.0 release, and so we should be allowed to make breaking changes.

I am happy to take a call with your team at Microsoft to explain why its nice for us to follow Microsoft's JSON API styleguide.

Reply:
@csuwildcat we had many calls on this, and I have documented my position clearly many times, both verbally and in writing.... one more time:

My preferences are:

1. Sidetree has a JSON API styleguide that it uses to make decisions
2. The styleguide recommends camelCase (among other things, see Microsoft and Google JSON API style guides)

Reply:
My default stance is that snake casing was decided by consensus vote, so we now need to stick with it, as multiple implementers are down the road and relying on it. My comment was more expressing my frustration from months back, but not suggesting we change it, certainly not in this PR.

Reply:
@thehenrytsai can you find a styleguide that aligns with this decision and add it to the README?

Reply:
Since this discussion is progressing in slack, I will resolve this comment. We can make an issue for having a style guide and bring it up in a WG meeting if we choose to do so :)